llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7570,10 +7574,10 @@ VectorizationFactor LoopVectorizationPlanner::computeBestVF() {
   // Set PlanForEarlyExitLoop to true if the BestPlan has been built from a
   // loop with an uncountable early exit. The legacy cost model doesn't
   // properly model costs for such loops.
+  auto ExitBlocks = BestPlan.getExitBlocks();
   bool PlanForEarlyExitLoop =
-      BestPlan.getVectorLoopRegion() &&
-      BestPlan.getVectorLoopRegion()->getSingleSuccessor() !=
-          BestPlan.getMiddleBlock();
+      ExitBlocks.size() > 1 ||
+      (ExitBlocks.size() == 1 && ExitBlocks[0]->getNumPredecessors() > 1);
```

Comment:
Yeah I agree with @ayalz, this does look like an independent change. Is it worth a separate NFC patch?